core/controllers/practice_sessions.py

```diff
@@ -32,12 +33,30 @@ def get(self, topic_name):
 
         topic = topic_services.get_topic_by_name(topic_name)
 
+        if topic is None:
+            raise self.PageNotFoundException(
+                Exception('The topic with the given name doesn\'t exist.'))
+
+        self.render_template('/pages/practice_session/practice_session.html')
+
+
+class PracticeSessionsPageDataHandler(base.BaseHandler):
+    """Fetches relevant data for the practice sessions page."""
+
+    GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
+
+    @acl_decorators.can_access_topic_viewer_page
+    def get(self, topic_name):
```

Comment:
Shouldn't this be hidden behind the ENABLE_NEW_STRUCTURE_PLAYERS flag?

Reply:
You're right, modified that Thanks!